### PR TITLE
Fixed bugs with the tab-order-sorting

### PR DESCRIPTION
**Pull request by OddDev**
0 based tab indexes are put AFTER positive valued ones
negative valued tab indexes can not be focused at all
Fixed both of these behaviors
Added tests to reflect these changes

### CODE REVIEW

File: src/useFocusTrap.js [hunk 1]
```diff
@@ -63,14 +63,11 @@ export function useFocusTrap() {
         const shiftKeyPressed = !!evt.shiftKey;
         let focusableElems = Array.from(
           trapRefElem.querySelectorAll(focusableElementsSelector)
+        ).filter(
+          (focusableElement) => focusableElement.getAttribute("tabindex") >= 0
```
Comment (activenode) [inline src/useFocusTrap.js]: I think this one is a potential troublemaker. What if no `tabindex` is defined e.g. `a[href]` ?

File: src/util.js [hunk 2]
```diff
@@ -2,3 +2,29 @@ export function convertToIntOrFallback(stringToConvert) {
   const parsed = parseInt(stringToConvert);
   return parsed ? parsed : 0;
 }
+
+export function sortByTabIndex(firstNode, secondNode) {
+  const tabIndexes = [firstNode, secondNode].map((node) =>
```
Comment (activenode) [inline src/util.js:7]: `tabIndices`

File: src/util.js [hunk 3]
```diff
@@ -2,3 +2,29 @@ export function convertToIntOrFallback(stringToConvert) {
...
+    getTabIndexOfNode(node)
+  );
+  return tabIndexes
+    .map((tabIndexValue) =>
+      sanitizeTabIndexInput(tabIndexValue, Math.max(...tabIndexes))
+    )
+    .reduce((previousValue, currentValue) => previousValue - currentValue);
+}
+
+function sanitizeTabIndexInput(tabIndex, maxValue) {
```
Comment (activenode) [inline src/util.js]: This function is hard to understand within this context. Can you add a comment?

File: src/util.js (hunk 2 above)
Comment (OddDev) [reply src/util.js:7]: Indexes are fine according to Linguee

File: src/useFocusTrap.js (hunk 1 above)
Comment (OddDev) [reply src/useFocusTrap.js]: f3dcf1fbe6b487c64ab44c819341d19fb1fdd162

File: src/util.js (hunk 3 above)
Comment (OddDev) [reply src/util.js]: df6e776e2d2efba48d6eecb1f7e32ab56c85fd7b

**Comment by OddDev** [comment]
Fixed first review iteration

**Pull request merged**